src/libs/actions/templates.ts

```diff
@@ -164,6 +164,51 @@ export async function addGeneratedTemplateHistory({
   }
 }
 
+export async function fetchUsageMetrics(
+  email: string,
+  page: number = 1,
+  pageSize: number = 10,
+): Promise<PaginatedResponse> {
+  try {
+    if (!email) {
+      throw new Error ('Please Provide email id');
+    }
+    const offset = (page - 1) * pageSize;
+
+    const totalRecords = await db
```

Comment:
dont create a new query for this, use the same query used to fetch metrics and perform a count aggregate on that query